runtime/src/main/java/org/corfudb/runtime/view/TableRegistry.java

```diff
@@ -262,7 +347,13 @@ private void insertAllDependingFileDescriptorProtos(TableDescriptors.Builder tab
             }
 
             // Add the fileDescriptorProto into the tableDescriptor map.
-            tableDescriptorsBuilder.putFileDescriptors(fileDescriptorProto.getName(), fileDescriptorProto);
+            tableDescriptorsBuilder.putFileDescriptors(fileDescriptorProto.getName(),
+                    FileDescriptorProto.getDefaultInstance());
```

Comment:
this is the key part which does the de-duplication.  Earlier we had the same file descriptor proto for every protobuf in the same file.  Now this is replacing it with a generic protobuf with default values.  Is that right?

Reply:
exactly default values take up almost nothing compared to the real thing and that's where the space savings come from.